### PR TITLE
Improve diagnostic for missing space in range pattern

### PR DESCRIPTION
**Pull request by clubby789**
Improves the diagnostic in #107425 by turning it into a note explaining the parsing issue.

r? @compiler-errors

### CODE REVIEW

File: compiler/rustc_parse/src/parser/pat.rs [hunk 1]
```diff
@@ -789,7 +789,7 @@ impl<'a> Parser<'a> {
 
     fn error_inclusive_range_match_arrow(&self, span: Span) {
         let after_pat = span.with_hi(span.hi() - rustc_span::BytePos(1)).shrink_to_hi();
-        self.sess.emit_err(InclusiveRangeMatchArrow { span, after_pat });
+        self.sess.emit_note(InclusiveRangeMatchArrow { span, after_pat });
```
Comment (compiler-errors) [inline compiler/rustc_parse/src/parser/pat.rs:792]: I think this should still be an error, and keep the primary message that explains the error. 

The "this is parsed as an inclusive range" should just be a note (or probably a *label*) appended to this error.

File: tests/ui/half-open-range-patterns/half-open-range-pats-inclusive-match-arrow.stderr [hunk 2]
```diff
@@ -15,5 +15,5 @@ error: expected one of `=>`, `if`, or `|`, found `>`
 LL |         74..=> {},
    |              ^ expected one of `=>`, `if`, or `|`
```
Comment (compiler-errors) [inline tests/ui/half-open-range-patterns/half-open-range-pats-inclusive-match-arrow.stderr:16]: You could also probably suppress this error (delaying it as a bug, perhaps) by checking if the previous token is `..=` in the match arm parsing code?

File: compiler/rustc_parse/src/parser/expr.rs [hunk 3]
```diff
@@ -2707,6 +2707,14 @@ impl<'a> Parser<'a> {
                     );
                     err.emit();
                     this.bump();
+                } else if matches!(
+                    (&this.prev_token.kind, &this.token.kind),
+                    (token::DotDotEq, token::Gt)
+                ) {
+                    // `error_inclusive_range_match_arrow` handles cases like `0..=> {}`,
+                    // so we supress the error here
+                    err.cancel();
```
Comment (estebank) [inline compiler/rustc_parse/src/parser/expr.rs:2716]: can you instead make this `err.delay_as_bug()`? That ensures that if this line is ever reached and no other error is emitted, the compile won't succeed. It's a failsafe against future regressions.

**Review by estebank — APPROVED** [review]

**Pull request merged**